code/modules/core_implant/cruciform/machinery/eotp.dm

```diff
@@ -0,0 +1,196 @@
+			eotp.removeObservation(10)
+
+		scanned.Remove(H)
+		last_rescan = world.time
+
+	updatePower()
+
+/obj/machinery/power/eotp/proc/addObservation(var/number)
+	observation += number
+	return observation
+
+/obj/machinery/power/eotp/proc/removeObservation(var/number)
+	observation -= number
+	return observation
+
+/obj/machinery/power/eotp/proc/updateObservation()
+	if(observation > max_observation)
+		observation = max_observation
+
+	if(observation < min_observation)
+		observation = min_observation
+
+/obj/machinery/power/eotp/proc/updatePower()
+	power_gaine = initial(power_gaine) + (observation / 100)
+
+	if(world.time >= (last_power_update + power_cooldown))
+		power += power_gaine
+		last_power_update = world.time
+
+	if(power >= max_power)
+		power -= max_power
+		power_release()
+
+/obj/machinery/power/eotp/proc/power_release()
+	var/type_release = pick(rewards)
+
+	if(type_release == ARMAMENTS)
+		var/reward_disk = pick(disk_types)
+		var/obj/item/_item = new reward_disk(get_turf(src))
+		visible_message(SPAN_NOTICE("The [_item.name] appers out of bluespace near the [src]!"))
+
+	else if(type_release == ALERT)
+
+		var/area/antagonist_area
+		var/preacher
+
+		for(var/datum/antagonist/A in GLOB.current_antags)
+			if((A.id == ROLE_CARRION) ||(A.id == ROLE_BLITZ) || (A.id == ROLE_BORER))
+				var/mob/living/L = A.owner.current
+				if(!isghost(L))
+					antagonist_area = get_area(L)
+					break
+		if(!antagonist_area)
+			for(var/disciple in disciples)
+				to_chat(disciple, SPAN_NOTICE("You can sleep safely. God protects you with his eye."))
+				if(ishuman(disciple))
+					var/mob/living/carbon/human/H = disciple
+					if(H.sanity)
+						H.sanity.changeLevel(20)
+			return
+
+		for(var/disciple in disciples)
+			if(ishuman(disciple))
+				var/mob/living/carbon/human/H = disciple
+				if(H.mind)
+					var/assigned_job = H.mind.assigned_job
+					if(istype(assigned_job, /datum/job/chaplain))
+						preacher = H
+
+		if(!preacher)
+			preacher = pick(disciples)
+
+		to_chat(preacher, SPAN_DANGER("You feel the evil in [antagonist_area]!"))
+
+	else if(type_release == INSPIRATION)
+		for(var/disciple in disciples)
+			if(ishuman(disciple))
+				var/mob/living/carbon/human/H = disciple
+				if(H.sanity)
+					if(prob(50))
+						H.sanity.positive_breakdown()
+
+	else if(type_release == ODDITY)
+		var/oddity_reward = pick(subtypesof(/obj/item/weapon/oddity/nt))
+		var/obj/item/_item = new oddity_reward(get_turf(src))
+		visible_message(SPAN_NOTICE("The [_item.name] appers out of bluespace near the [src]!"))
+
+	else if(type_release == STAT_BUFF)
+		var/random_stat = pick(ALL_STATS)
+		for(var/disciple in disciples)
+			if(ishuman(disciple))
+				var/mob/living/carbon/human/H = disciple
+				if(H.stats)
+					to_chat(H, SPAN_NOTICE("You feel the gaze of the [src] pierce your mind, body, and soul. You are enlightened, and gain deeper knowledge in [random_stat]; however, you can already feel this newfound knowledge beginning to slip away.."))
```

Comment:
I'm pretty sure this is going to say 'the the eye of the protector', so lemme fix that 
```suggestion
					to_chat(H, SPAN_NOTICE("You feel the gaze of [src] pierce your mind, body, and soul. You are enlightened, and gain deeper knowledge in [random_stat]; however, you can already feel this newfound knowledge beginning to slip away.."))
```